libs/vkd3d-common/debug.c

```diff
@@ -281,11 +285,14 @@ const char *debugstr_w(const WCHAR *wstr)
 
 unsigned int vkd3d_env_var_as_uint(const char *name, unsigned int default_value)
 {
-    const char *value = getenv(name);
+    char value[VKD3D_PATH_MAX];
+
+    value[0] = '\0';
+    vkd3d_get_env_var(name, value, sizeof(value));
```

Comment:
vkd3d_get_env_var should ensure the string remains in a sane state. Looks cleaner than sprinkling `[0] = '\0'` code everywhere.

Reply:
On entry to get_env, I guess we can assert that size is > 0, so:

```
assert(size > 0);
str[0] = '\0';
do_stuff();
```